sympy/simplify/radsimp.py

```diff
@@ -254,7 +257,7 @@ def parse_expression(terms, pattern):
         terms is a list of tuples as returned by parse_terms;
         pattern is an expression treated as a product of factors
         """
-        pattern = Mul.make_args(pattern)
+        pattern = list(ordered(Mul.make_args(pattern)))
```

Comment:
This will be redundant in my branch, where args are ordered by default_sort_key.